apps/sync/js/launcher.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global
+  LazyLoader,
+  SyncEngine
+*/
+
+/* exported
+  Launcher,
+  DataAdapters
+*/
+
+var DataAdapters = {
+  // To be filled by js/adapters/*.js
+};
+
+const Launcher = Object.freeze((() => {
+  var running = false;
+
+  const loadMainScripts = () => {
+    if (typeof SyncEngine !== 'undefined') {
+      return Promise.resolve();
+    }
+    return LazyLoader.load([
+      'js/crypto/stringconversion.js',
+      'js/crypto/keyderivation.js',
+      'js/crypto/fxsyncwebcrypto.js',
+
+      // to be added in https://bugzilla.mozilla.org/show_bug.cgi?id=1206615:
+      'js/ext/kinto.dev.js',
+      'js/sync-engine/syncengine.js'
+    ]);
+  };
+
+  const loadDataAdapterScript = (collectionName) => {
+    if (typeof DataAdapters[collectionName] !== 'undefined') {
+      return Promise.resolve();
+    }
+    return LazyLoader.load(`js/adapters/${collectionName}.js`);
+  };
+
+  return {
+    /**
+      * @param request {Object} Should contain the following fields:
+      *                         * assertion {String} A BrowserID assertion.
+      *                         * keys {Object} Like `{ kA: 'foo', kB: 'bar' }`.
+      *                         * collections {Object} Like `{ history: {} }`.
+      *                         * URL {String} Like 'http://localhost:8000/v1/'.
+      */
+    handleSyncRequest(request) {
+      if (running) {
+        return Promise.reject(new Error('A sync request is already running.'));
```

Comment:
Add the errors to [1] and follow the same naming convention, please. (i.e. ERROR_SYNC_APP_SYNC_IN_PROGRESS)

[1] https://mxr.mozilla.org/gaia/source/shared/js/sync/errors.js